jest.config.ts

```diff
@@ -0,0 +1,67 @@
+import type { Config } from 'jest';
+
+const config: Config = {
+  // Automatically clear mock calls, instances, contexts and results before every test
+  clearMocks: true,
+
+  // Indicates whether the coverage information should be collected while executing the test
+  collectCoverage: true,
+
+  // The directory where Jest should output its coverage files
+  coverageDirectory: 'coverage',
+
+  // Indicates which provider should be used to instrument code for coverage
+  coverageProvider: 'babel',
+
+  // A map from regular expressions to module names or to arrays of module names that allow to stub out resources with a single module
+  moduleNameMapper: {
+    '^(\\.{1,2}/.*)\\.js$': '$1', // Handle ESM imports
+  },
+
+  // A preset that is used as a base for Jest's configuration
+  preset: 'ts-jest',
+
+  // Run tests from one or more projects
+  projects: [
+    {
+      displayName: 'jsdom',
+      testEnvironment: 'jsdom',
+      testMatch: ['**/__tests__/**/*.spec.tsx'],
+      transform: {
+        '^.+\\.tsx?$': 'ts-jest', // Use ts-jest for TypeScript files
+      },
+      moduleNameMapper: {
+        '^(\\.{1,2}/.*)\\.js$': '$1',
+      },
+    },
+    {
+      displayName: 'node',
+      testEnvironment: 'node',
+      testMatch: ['**/__tests__/**/*.spec.ts'],
```

Comment:
Any specific reason why? I find that it clutters up the source file somewhat by having the tests in there in parallel.

Reply:
My big issue is that the `src` folder is already pretty cluttered since we dump all files in there without any subdivision. This will be fixed in https://github.com/workos/authkit-nextjs/pull/147 however where I've at least separated the source by components.

I'm going to merge this PR as is for now, but once the mentioned PR has merged I'll move the tests to be alongside the rest of the code since it would make it more consistent with other authkit repos as you mentioned.